contracts/staking-vault/StakingVault.sol

```diff
@@ -0,0 +1,284 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin-4.8.0/token/ERC1155/ERC1155.sol";
+import "@openzeppelin-4.8.0/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin-4.8.0/token/ERC20/extensions/IERC20Metadata.sol";
+
+import "../../interfaces/IPriceFeeds.sol";
+import "../proxies/0_8/Upgradeable_0_8.sol";
+
+contract StakingVault is Upgradeable_0_8, ERC1155 {
```

Comment:
since you have interface maybe its wroth inheriting it to enforce `interface StakingVault`

Reply:
agreed

Reply:
implemented